collect_app/src/main/java/org/odk/collect/android/preferences/PreferenceVisibilityHandler.kt

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2017 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.odk.collect.android.preferences
+
+import androidx.preference.PreferenceGroup
+import androidx.preference.PreferenceScreen
+import org.odk.collect.android.preferences.keys.ProjectKeys
+import org.odk.collect.android.preferences.keys.ProtectedProjectKeys
+import org.odk.collect.android.preferences.source.SettingsProvider
+import org.odk.collect.android.version.VersionInformation
+
+class PreferenceVisibilityHandler(
```

Comment:
It might seem like we don't have test for this class but in fact everything is tested in fragment tests like: `FormManagementPreferencesFragmentTest`